tests/test.c

```diff
@@ -649,3 +663,52 @@ void check_param_range(char* dev_name, uint64_t uid, char* param_name, param_typ
     }
     print_pass();
 }
+
+// Returns if the latency is neither too high nor negative
+void check_latency(uint64_t uid, int32_t upper_bound_latency, uint64_t start_time) {
+    char* param_name = "TIMESTAMP";
+    uint8_t dev_type = device_name_to_type("TimeTestDevice");
+    if(dev_type == -1){
+        print_fail();
+        fprintf(stderr, "TimeTestDevice type not found\n");
+        end_test();
+        exit(1);
+    }
+    device_t* dev = get_device(dev_type);
+    if(dev == NULL){
+        print_fail();
+        fprintf(stderr, "Type: %d not found\n", dev_type);
+        end_test();
+        exit(1);
+    }
+    param_val_t vals_after[dev->num_params];
+    int8_t param_idx = get_param_idx(dev_type, param_name);
+    if(param_idx < 0) {
+        print_fail();
+        fprintf(stderr, "Could not find param_idx\n");
+        end_test();
+        exit(1);
+    }
+    int success = device_read_uid(uid, EXECUTOR, DATA, (1 << param_idx), vals_after);
+    if(success < 0) {
+        fprintf(stderr, "Error reading device with UID: %llu", uid);
+    }
+    
+    // shorten to last 9 digits, like the TimeTestDevice's TIMESTAMP
+    uint32_t start_time_shortened = start_time  % 1000000000; 
+    param_val_t end_time = vals_after[param_idx];
+    
+    int32_t elapsed_time = end_time.p_i - start_time_shortened;
+    fprintf(stderr, "Elapsed time is %d\n", end_time.p_i);
+    if(elapsed_time >= upper_bound_latency || elapsed_time < 0){
```

Comment:
if `elapsed_time < 0`, doesn't it mean there is some other error going on? we should handle it differently to print out a different error msg.

Reply:
Hmm yea I see your point. Let me fix that real quick